lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -33,17 +37,92 @@ describe('NxModal', function() {
 
   it('merges any passed in className to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal className="test"/>
+      <NxModal className="test" onClose={dummyCloseHandler}/>
     );
     const nxModalDiv = nxModal.find('.nx-modal');
     expect(nxModalDiv.hasClass('test')).toEqual(true);
   });
 
   it('includes any passed in attributes to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal id="modal-id" data-foo="bar"/>
+      <NxModal id="modal-id" data-foo="bar" onClose={dummyCloseHandler}/>
     );
     expect(nxModal.find('.nx-modal').prop('id')).toEqual('modal-id');
     expect(nxModal.find('.nx-modal').prop('data-foo')).toEqual('bar');
   });
+
+  describe('NxModal event listener support', () => {
+    let containerMainModal: HTMLDivElement | null;
+    let containerSecondaryModal: HTMLDivElement | null;
+
+    beforeEach(function () {
+      // Rendering containerMainModal for the component in test.
+      containerMainModal = document.createElement('div');
+      containerSecondaryModal = document.createElement('div');
+
+      document.body.appendChild(containerMainModal);
+      document.body.appendChild(containerSecondaryModal);
+    });
+
+    afterEach(function () {
+      if (containerMainModal) {
+        document.body.removeChild(containerMainModal);
+        containerMainModal = null;
+      }
+
+      if (containerSecondaryModal) {
+        document.body.removeChild(containerSecondaryModal);
+        containerSecondaryModal = null;
+      }
+    });
+
+    it('executes onClose method when pressing ESC key', function () {
+      const mockCallBack = jest.fn();
+      const nxModal = <NxModal id='first-modal-id' onClose={mockCallBack}/>;
```

Comment:
Minor: we use double quotes in HTML and JSX

```suggestion
      const nxModal = <NxModal id="first-modal-id" onClose={mockCallBack}/>;
```

Reply:
> Minor: we use double quotes in HTML and JSX

Use double quotes on JSX: a88dab1